x-pack/plugins/apm/server/routes/alerts/rule_types/anomaly/register_anomaly_rule_type.ts

```diff
@@ -250,6 +254,7 @@ export function registerAnomalyRuleType({
                 transactionType: latest.by_field_value as string,
                 environment: job.environment,
                 score: latest.record_score as number,
+                detectorIndex: latest.detector_index as number,
```

Comment:
I wonder if it's better to store the detector type than the index. Say, we change the order of detectors (unlikely and probably already breaking).
Haven't made my mind up but wanted to throw it out there in case someone has an opinion.

Reply:
We have tests for that, and will break in several places not only anomaly rules.

Reply:
It's also about what's more readable. Storing an index is quite opaque. Eg in a couple of months when debugging an alert will you recall which detector index is what? Will other people?

Reply:
++ on storing the detector type. the detector index is unique to the job, so if we e.g. add a job for span metrics, the detector index will be re-used across jobs.

Reply:
Fixed here https://github.com/elastic/kibana/pull/171901/commits/b750f0d3e2738af7cfd51435eaf7cca01036781d